### ISSUE TITLE
Increasing saturation of gray lead to pink

### ISSUE THREAD

**Issue by giagitom** [molstar/molstar#842]
I noticed this by tuning the element symbol saturation adjustment setting. Carbon is gray by default.

### saturation = 0
![image](https://github.com/molstar/molstar/assets/20627977/20ddeca9-2174-4818-b7a0-168f870fe7fb)

### saturation = 2.6
![image](https://github.com/molstar/molstar/assets/20627977/56b3cfd6-df05-4fe6-894e-34f989ba3b7a)

**Comment by arose** [comment]
I am not really a color space expert to know if that is expected behavior or not. We saturate in HCL space and do conversion to/from sRGB.

**Comment by giagitom** [comment]
> I am not really a color space expert to know if that is expected behavior or not.
Me neither, it just appeared strange to me.

I'm preventing this in the following way on my side. (I'm using HSL color space, but I guess it could be similar with HCL)

```js
      const hsl = rgbToHsl(hexToRgb(hex))
      if (hsl.s === 0 && saturation > 0) hsl.s = 0.0001
      else hsl.s = Math.max(0,Math.min(hsl.s + saturation, 1))
      return rgbToHex(hslToRgb(hsl))
```

**Comment by arose** [comment]
OK, had a chance to think about it and grey/white/black/... don't have a hue, so there is a) nothing to saturate/desaturate and b) when you convert it to HCL the H (hue) gets set to NaN which we don't handle and that is what lads to the pink color.

I plan to change it so that Color.saturate/.desaturate will leave colors without hue (i.e., r == g == b) unchanged.

**Comment by arose** [comment]
fixed in d5a47e617a515a4afca892bd45fcbfcfa94b84fb

**Comment by giagitom** [comment]
Awesome, Thanks!
